main/auth/external_login/ldap.inc.php

```diff
@@ -405,3 +411,92 @@ function extldap_add_user_by_array($data, $update_if_exists = true)
 
     return $user_id;
 }
+
+/**
+ * Get one user's single attribute value.
+ * User is identified by filter.
+ * $extldap_config['filter'] is also applied in complement, if defined.
+ *
+ * @param $filter string LDAP entry filter, such as '(uid=10000)'
+ * @param $attribute string name of the LDAP attribute to read the value from
+ * @return string|bool the single matching user entry's single attribute value or false if not found
+ * @throws Exception if more than one entries matched or on internal error
+ */
+function extldap_get_user_attribute_value($filter, $attribute)
+{
+    global $extldap_config;
+
+    if (array_key_exists('filter', $extldap_config) && !empty($extldap_config['filter'])) {
+        $filter = '(&'.$filter.'('.$extldap_config['filter'].'))';
+    }
+
+    $ldap = extldap_connect();
+    if (false === $ldap) {
+        throw new Exception(get_lang('LDAPConnectFailed'));
+    }
+
+    if (false === ldap_bind($ldap, $extldap_config['admin_dn'], $extldap_config['admin_password'])) {
+        throw new Exception(get_lang('LDAPBindFailed'));
+    }
+
+    $searchResult = ldap_search($ldap, $extldap_config['base_dn'], $filter, [ $attribute ]);
+    if (false === $searchResult) {
+        throw new Exception(get_lang('LDAPSearchFailed'));
+    }
+
+    switch(ldap_count_entries($ldap, $searchResult)) {
+        case 0:
+            return false;
+        case 1:
+            $entry = ldap_first_entry($ldap, $searchResult);
+            if (false === $entry) {
+                throw new Exception(get_lang('LDAPFirstEntryFailed'));
+            }
+            $values = ldap_get_values($ldap, $entry, $attribute);
+            if (false == $values) {
+                throw new Exception(get_lang('LDAPGetValuesFailed'));
+            }
+            if ($values['count'] == 1) {
+                return $values[0];
+            }
+            throw new Exception(get_lang('MoreThanOneAttributeValueFound'));
+        default:
+            throw new Exception(get_lang('MoreThanOneUserMatched'));
+    }
+}
+
+/**
+ * Get the username from the CAS-supplied user identifier
+ *
+ * searches in attribute $extldap_user_correspondance['extra']['cas_user'] or 'uid' by default
+ * reads value from attribute $extldap_user_correspondance['username'] or 'uid' by default
+ *
+ * @param $casUser string code returned from the CAS server to identify the user
+ * @return string|bool user login name, false if not found
+ * @throws Exception on error
+ */
+function extldap_cas_user_login($casUser)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The variable $extldap_user_correspondance is not named in camelCase.](https://app.codacy.com/manual/chamilo/chamilo-lms/pullRequest?prid=4906847)

Reply:
global variable $extldap_user_correspondance was already defined and used elsewhere.